### PR TITLE
Fix issue reported by `gosec` and enable `gosec` in the CI

### LINKED ISSUE #4127: Fix the HIGH and Medium issues reported by gosec
## Feature Request

**Is your feature request related to a problem? Please describe:**

Fix the HIGH and Medium issues reported by gosec.
After the fix, enable `gosec` in the golangci linters to avoid introducing High or Medium issues later.

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Close #4127

### Code changes

- [x] Has Go code change
- [x] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: tests/e2e/e2e.go [hunk 1]
```diff
@@ -172,7 +172,7 @@ func setupSuite(c kubernetes.Interface, extClient versioned.Interface, apiExtCli
 		if localStorageSC == nil {
 			log.Fail("local-storage storage class not found")
 		}
-		if localStorageSC.Annotations == nil {
+		if localStorageSC != nil && localStorageSC.Annotations == nil {
```
Comment (csuzhangxc) [inline tests/e2e/e2e.go:175]: we already have `if localStorageSC == nil {` in L172?

**Review by csuzhangxc — APPROVED** [review]

File: tests/e2e/e2e.go (hunk 1 above)
Comment (KanShiori) [reply tests/e2e/e2e.go:175]: yes, but `gosec` still report this issue. 😂

Comment (csuzhangxc) [reply tests/e2e/e2e.go:175]: nolint?

Comment (july2993) [reply tests/e2e/e2e.go]: I think it's because log.Fail is not guaranteed to quit statically

Comment (july2993) [reply tests/e2e/e2e.go]: we can just return after L173

Comment (KanShiori) [reply tests/e2e/e2e.go]: got it

File: pkg/manager/member/dm_master_upgrader_test.go [hunk 2]
```diff
@@ -263,8 +263,8 @@ func TestMasterUpgraderUpgrade(t *testing.T) {
 		},
 	}
 
-	for _, test := range tests {
-		testFn(&test)
+	for i := range tests {
```
[the remark this answers is not in the document]
Comment (KanShiori) [reply pkg/manager/member/dm_master_upgrader_test.go:266]: yes, but I think this usage is safer.

**Review by july2993 — APPROVED** [review]

**Pull request merged**